compiler/src/main/scala/org/scalajs/nscplugin/PrepJSExports.scala

```diff
@@ -531,9 +516,28 @@ trait PrepJSExports[G <: Global with Singleton] { this: PrepJSInterop[G] =>
     }
 
     // Construct proxied function call
-    val sel = gen.mkAttributedRef(trgSym)
+    val sel = {
+      if (trgSym.isConstructor) {
+        Select(New(TypeTree(trgSym.owner.tpe)), trgSym)
+      } else if (trgSym.isModuleClass) {
+        assert(proxySym.paramss.isEmpty,
+            s"got a module export with non-empty paramss. target: $trgSym, proxy: $proxySym")
+        gen.mkAttributedRef(trgSym.sourceModule)
+      } else if (trgSym.isTrait) {
+        // We'll emit an error, but generate a tree to avoid crashing.
+        EmptyTree
+      } else if (trgSym.isClass) {
+        assert(isJSAny(trgSym), s"got a class export for a non-JS class ($trgSym)")
+        assert(proxySym.paramss.isEmpty,
+            s"got a class export with non-empty paramss. target: $trgSym, proxy: $proxySym")
+        TypeApply(gen.mkAttributedRef(JSPackage_constructorOf), List(TypeTree(trgSym.tpe)))
```

Comment:
This might also create an inconsistent tree if the class is polymorphic. We might need to use `trgSym.tpe_*` or something like that. We have precedent in `ExplicitLocalJS`:
https://github.com/scala-js/scala-js/blob/225395005a56cf7c2a55a88f15ba689dd36aa76f/compiler/src/main/scala/org/scalajs/nscplugin/ExplicitLocalJS.scala#L237

Reply:
Note to self: I have gotten this to work with `tpe_*`, but I'm not sure it is a good idea to leave a `tpe_*` traverse so many compiler steps. I should investigate constructing the proper type.